doc/design/design_model_zoo_pai.md

```diff
@@ -0,0 +1,97 @@
+# Model Zoo on PAI TensorFlow
+
+## Introduction
+
+[Alibaba PAI](https://www.alibabacloud.com/product/machine-learning) is an end-to-end platform that provides various machine learning algorithms to meet users' data mining and analysis requirements. PAI supports TensorFlow and many other machine learning frameworks. SQLFlow model zoo also works on PAI. [Model Zoo Design Doc](./design_model_zoo.md) is a high level design for SQLFlow model zoo. This document is about how to do model training, model prediction, and model analysis using SQLFlow model zoo on PAI.
+
+## Background
+### Submit a Task to PAI
+A PAI program is a normal program in TensorFlow or other framework, except that PAI requires its application expose several command line options to communicate with [MaxCompute](https://www.alibabacloud.com/product/maxcompute) and OSS([Alibaba Cloud Object Storage Service](https://www.alibabacloud.com/product/oss)). Typically, a user submit a PAI task with one of the following methods:
+1. Use the MaxCompute CLI `odpscmd`:
+```bash
+$ odpscmd -e "pai -name tensorflow -Dscript=file:///my/training/script.py -Dother_param=other_param_value..."
+
+```
+When a user runs the command above, the CLI will submit the task to PAI then print a logview URL and an instance ID to stdout, after which the user can safely terminate the CLI. The user can check the logview to get the running status of the task or run:
+```bash
+$ odpscmd -e "wait instanceID"
+```
+to attach to the running task.
+
+
+2. Use the MaxCompute Python Package `pyodps`:
+```python
+
+from odps import ODPS
+
+conn = ODPS(...)
+inst = conn.run_xflow('tensorflow', parameters={'script':      'file:///my/training/script.py',
+               	                                'other_param': 'other_param_value' ... })
+print(instance.get_logview_address())
+```
+This code snippet asynchronously submits a task to PAI and prints the logview URL to stdout.
+
+### Train/Predict/Analyze in SQLFlow with Model Zoo
+In SQLFlow, we train a model with the following SQLFlow extened SELECT statement:
+```sql
+SELECT * FROM train_table ... TO TRAIN MyAwesomeEstimator ... INTO my_awesome_model;
+```
+and predict with:
+```sql
+SELECT * FROM pred_table ... TO PREDICT using my_awesome_model ... ;
+```
+and analyze with:
+```sql
+SELECT * FROM train_table ...  ANALYZE my_awesome_model USING TreeExplainer;
+
+```
+The extended SELECT statements are executed as described in [Model Zoo Design Doc](./design_model_zoo.md#Submitter-Programs):
+- If the model **is not** from a model zoo, The SQLFlow server converts these statements to a *submitter* file and submits the file to a specified engine.
+- If the model **is** from a model zoo, the submmiter program is mapped into a Docker image. the SQLFlow server calls k8s API to launch the Docker image on a k8s cluster.
```

Comment:
k8s is always needed?

Reply:
I remember we talked about using minikube for single machine users in the near future, so shouldn't we only consider k8s here?

Reply:
The PAI Tensorflow training job runs on the cluster resource provided by PAI. Maybe `sqlflowserver` is deployed on k8s, the argo workflow runs on k8s, but the training job runs on PAI.

Reply:
the SQLFlow server -> The SQLFlow server